include/openvic-dataloader/detail/SymbolIntern.hpp

```diff
@@ -1,14 +1,266 @@
 #pragma once
 
 #include <cstdint>
+#include <iterator>
+#include <string_view>
+
+#include <openvic-dataloader/detail/pinned_vector.hpp>
 
 #include <dryad/symbol.hpp>
 
 namespace ovdl {
+	// Contains all unique symbols, null-terminated, in memory one after the other.
+	template<typename CharT>
+	struct symbol_buffer {
+		static constexpr auto min_buffer_size = 16 * 1024;
+
+		constexpr symbol_buffer() : _data_buffer(ovdl::detail::max_elements(min_buffer_size + 1)) {}
+		explicit symbol_buffer(std::size_t max_elements)
+			: _data_buffer(ovdl::detail::max_elements(std::max<std::size_t>(max_elements, min_buffer_size + 1))) {
+			_data_buffer.reserve(min_buffer_size);
+		}
+
+		void free() {
+		}
```

Comment:
Is this still needed? Can it be removed along with the only call to it later in this file?